website/server/libs/payments/subscriptions.js

```diff
@@ -80,19 +80,9 @@ async function createSubscription (data) {
   let emailType = 'subscription-begins';
   let recipientIsSubscribed = recipient.isSubscribed();
 
-  if (data.user && !data.gift && !data.groupId) {
-    const unlockedUser = await User.findOneAndUpdate(
-      {
-        _id: data.user._id,
-        $or: [
-          { _subSignature: 'NOT_RUNNING' },
-          { _subSignature: { $exists: false } },
-        ],
-      },
-      { $set: { _subSignature: 'SUB_IN_PROGRESS' } },
-    );
-    if (!unlockedUser) {
-      throw new NotFound('User not found or subscription already processing.');
+  if (data.user && !data.gift && !data.groupId && data.customerId !== 'group-plan') {
+    if (moment().diff(data.user.purchased.plan.dateUpdated, 'minutes') < 3) {
```

Comment:
do we have times how long this function needs? 

I had it sometimes that for example a browser - triggers a 2nd call (not visible in the network overview of devtools) if the first one didnt respond in 90ish seconds - what if its the same here

it only triggers another one because it didn't respond faster

maybe thats the actual part we should look into 

but if its not the issue of a too slow response then something like this is sadly the better solution

Reply:
We haven't seen it in browsers, it's happening with Google Play subscriptions from the mobile app...
it'd be good to fix it on the app side, but we haven't been able to track it down yet. In the meantime, then, we can try this, as if we can at least stop the double processing on the server side we can proceed with transaction logs and such!